Userland/Games/2048/Game.cpp

```diff
@@ -198,6 +219,72 @@ Game::MoveOutcome Game::attempt_move(Direction direction)
     return MoveOutcome::InvalidMove;
 }
 
+void Game::add_evil_tile()
+{
+    size_t worst_row = 0;
+    size_t worst_column = 0;
+    u32 worst_value = 2;
+
+    size_t most_free_cells = m_grid_size * m_grid_size; // a big value to min() against
+
+    // Largest theoretically possible score after a move.
+    // All the cells are filled with the same tile, so after a move every tile merges.
+    size_t worst_score = m_score + largest_tile() * m_grid_size * m_grid_size;
+
+    bool stalled = false;
+
+    for (size_t row = 0; row < m_grid_size && !stalled; row++) {
+        for (size_t column = 0; column < m_grid_size && !stalled; column++) {
+            if (m_board[row][column] != 0)
+                continue;
+
+            for (u32 value : Array { 2, 4 }) {
+                Game saved_state = *this;
+                saved_state.m_board[row][column] = value;
+
+                if (is_stalled(saved_state.m_board)) {
+                    stalled = true;
```

Comment:
We're OK with reasonable `goto` usage, right?

In this case, it looks like it would be clearer to drop this variable and structure the code like this:

```cpp
    for (coords : board) {
        for (possible value : {2, 4}} {
            if (is_stalled()) {
                // (Evil laughter)
                goto found_worst;
            }
        }
    }
found_worst:
   m_board[worst_row][worst_column] = worst_value;
```

Reply:
I can live with that.